modules/ROOT/pages/6.5-release-notes.adoc

```diff
@@ -108,6 +108,15 @@ In previous versions of {productname}, an issue was identified in the close butt
 
 In {productname} 6.5, the markup was fixed by replacing the div element with a `<span>` element, aligning it with HTML guidelines. As a result, the updated close buttons now pass HTML markup checkers without throwing errors.
 
+=== Context menus and menu items were not constrained within the scrollable container if the parent is a Shadow Host
+//#TINY-9743
+
+In previous versions of {productname}, context menus and menu items would overflow beyond the editor's bounds. 
+
+To fix this issue {productname} 6.5 added support for the `ShadowDom` when the scrollable container is inside the `ShadowRoot` and is the parent of the Shadow Host.
+
+As a result context menus and menu items are now constrained within the scrollable container if the parent is a Shadow Host.
```

Comment:
```suggestion
=== Context menus and menu items were not constrained within the scrollable container if the parent node was a Shadow Root
//#TINY-9743

In previous versions of {productname}, context menus and menu items could overflow beyond the editor’s bounds when the parent node was a Shadow root. 

To fix this issue {productname} 6.5 added support for the Shadow Dom API when the scrollable container is inside a Shadow root and is the parent of the Shadow host.

As a result context menus and menu items are now constrained within the scrollable container if the parent is a Shadow host.
```

1. Sub-head copy-edit for tense consistency.

2. I remain unhappy with this clause:
	_when the scrollable container is inside a Shadow root and is the parent of the Shadow host_

	Parsing this carefully, the scrollable container
	* is inside a Shadow root **and**
	* is the parent of the Shadow host

	This confuses me.

	A Shadow root is, by definition, attached to the so-called Light tree at the Shadow host.

	I can’t, therefore, see how something _inside_ a Shadow root is simultaneously a parent to the Shadow host (which, in effect, encompasses the Shadow root).

Reply:
> when the scrollable container is inside a Shadow root and is the parent of the Shadow host

In the previous release, 6.4, the scrollable container ticket supported ShadowDom, but only when the scrollable container was also inside the Shadow Root.
And now in this ticket 9743, we added the support where the scrollable container is the parent of the Shadow Host.

Reply:
The shadow root in this case is the root element of the editor. This issue occurred when the shadow host (parent element) of the shadow root was a scrollable element. So the order goes Scrollable container =/> shadow host > shadow root > editor. There may have been some confusing wording around these two elements. @shanmen-tiny correct me if I'm wrong.

Reply:
### Parsing @shanmen-tiny to see if I understand

#### in TinyMCE 6.4
- scrollable containers
  * support the Shadow DOM API
    - _iff_ the scrollable container is a child node within the Shadow root tree

#### in TinyMCE 6.5
- scrollable containers
  * support the Shadow DOM API
    - if the scrollable container is a child node within the Shadow root tree _and_
    - if the scrollable container is the parent of the Shadow host node.

### Put another way

#### in TinyMCE 6.5
- scrollable containers
  * support the Shadow DOM API
    - in two discrete and mutually exclusive name-space settings:
      1. when the scrollable container is a child node within the Shadow root tree; or
      2. when the scrollable container is a parent node of the Shadow host node.

Reply:
That's correct @brian-forte